pkg/asset/machines/azure/machines.go

```diff
@@ -0,0 +1,107 @@
+// Package azure generates Machine objects for azure.
+package azure
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	machineapi "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/azure"
+	azureprovider "sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1alpha1"
+)
+
+const (
+	// TODO(flaper87): We're choosing to hardcode these values to make
+	// the environment more predictable. We expect there to a secret
+	// named `azure-credentials` and a cloud named `AzurePublicCloud` in
+	// the clouds file stored in this secret.
+	cloudsSecret          = "azure-credentials"
```

Comment:
`cloudsSecret`  and `cloudsSecretNamespace` don't seemed to be used. drop them?

Reply:
no, they should actually be used here. that's a bug, will fix 
https://github.com/openshift/installer/blob/39a280cd1a649d72f70cf07bf41b96bd67e32fd2/pkg/asset/machines/azure/machines.go#L88

Reply:
fixed